modules/internal/String.chpl

```diff
@@ -427,6 +427,75 @@ module String {
       return ret;
     }
 
+    /* This function takes an array of stings as input.
```

Comment:
I think it would be more appropriate to put this function in modules/standard/Path.chpl, since its behavior is dependent on paths rather than strings themselves (i.e. you aren't performing a character-by-character comparison, you are comparing based on file path and directory chunks)

Reply:
As the function aims at finding the longest common path prefix, thus we had performed word-by-byword comparison. It has been added to modules/standard/Path.chpl as mentioned by you.
Also, it is not like Python's feature as we have returned the common path in all the paths provided by the user and have not done it character by character. But the input is an array of strings here (Python has lists). Python feature returns invalid paths but this feature doe not.

Reply:
> Also, it is not like Python's feature as we have returned the common path in all the paths provided by the user and have not done it character by character. But the input is an array of strings here (Python has lists). Python feature returns invalid paths but this feature does not.

I think that's a reasonable change to the Python implementation.  However, it is different from the specification that was discussed when we first proposed this function - that means it would be best to bring this up for discussion to ensure that is an acceptable change to the community at large (instead of just me :) ).

I would recommend commenting on the trade-offs of this decision on the original issue and sending an email to the developers and users mailing lists asking for comment on the issue with this proposal in mind.  One such trade-off would be the handling of the file prefix test case I mentioned - it's still supportable if you think that's reasonable, but would require more effort to do so